apache-kafka-2/src/main/java/com/baeldung/kafka/message/ordering/serialization/JacksonDeserializer.java

```diff
@@ -0,0 +1,36 @@
+package com.baeldung.kafka.message.ordering.serialization;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.kafka.common.serialization.Deserializer;
+
+import java.util.Map;
+
+public class JacksonDeserializer<T> implements Deserializer<T> {
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    private Class<T> tClass;
+
+    public JacksonDeserializer(Class<T> tClass) {
+        this.tClass = tClass;
+    }
+
+    public JacksonDeserializer() {
+
+    }
+
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        this.tClass = (Class<T>) configs.get("value.deserializer.serializedClass");
```

Comment:
It took me a while to realise that `value.deserializer.serializedClass` was a custom configuration property key and not a built in one, can we extract a constant for this string and use that constant elsewhere so we can see how it ties together

Reply:
Added config.